.jenkins/pipelines/standalone/azure-sdk-tests.Jenkinsfile

```diff
@@ -27,20 +28,41 @@ pipeline {
     stages {
         stage("Cleanup files") {
             steps {
-                sh """
-                   ${JENKINS_SCRIPTS}/global/clean-temp.sh
-                   """
+                sh "${JENKINS_SCRIPTS}/global/clean-temp.sh"
+            }
+        }
+        stage("Checkout Pull Request") {
+            when {
+                expression { params.PULL_REQUEST_ID != "" }
+            }
+            steps {
+                cleanWs()
+                checkout([$class: 'GitSCM',
+                    branches: [[name: "pr/${PULL_REQUEST_ID}"]],
+                    extensions: [],
+                    userRemoteConfigs: [[
+                        url: 'https://github.com/deislabs/mystikos',
+                        refspec: "+refs/pull/${PULL_REQUEST_ID}/merge:refs/remotes/origin/pr/${PULL_REQUEST_ID}"
+                    ]]
+                ])
             }
         }
         stage('Verify commit sync') {
-            when { allOf {
+            when {
                 expression { params.COMMIT_SYNC != "" }
-                expression { params.COMMIT_SYNC != GIT_COMMIT }
-            }}
+            }
             steps {
+                // Check if the checked out commit is the same across all parallel builds
                 script {
-                    currentBuild.result = 'ABORTED'
-                    error("Aborting build: mismatched commit - commit($GIT_COMMIT), expected(${COMMIT_SYNC})")
+                    def GIT_COMMIT_ID = sh(
```

Comment:
Thoughts on using the build in `GIT_COMMIT` variable instead?

Reply:
I'd like to avoid reusing the built in `GIT_COMMIT` to avoid confusion. GIT_COMMIT is used for the commit for the initial Jenkinsfile scm checkout and in many cases GIT_COMMIT_ID will be the same as GIT_COMMIT. But for PR builds it will not be the same as we don't check out a PR head/merge for the initial Jenkisnfile checkout.